app/code/Magento/Shipping/Controller/Adminhtml/Order/Shipment/Save.php

```diff
@@ -118,8 +131,9 @@ public function execute()
             $this->shipmentLoader->setTracking($this->getRequest()->getParam('tracking'));
             $shipment = $this->shipmentLoader->load();
             if (!$shipment) {
-                $this->_forward('noroute');
-                return;
+                $resultForward = $this->resultForwardFactory->create();
```

Comment:
Looks like we can get resultForward by without getting one more dependency:

Here is example:
https://github.com/magento/magento2/blob/df2784b6dec6fe841c72223705974231c246a465/app/code/Magento/Catalog/Controller/Adminhtml/Product/Reload.php#L22

Reply:
@ihor-sviziev Updated the PR with this change.